pulumi/volumeReaper/src/volumeReaper.ts

```diff
@@ -0,0 +1,196 @@
+import * as k8s from '@kubernetes/client-node'
+import assert from 'assert'
+
+const delay = (time: number) => {
+  return new Promise((resolve) => setTimeout(resolve, time))
+}
+
+interface StatefulSet extends k8s.V1StatefulSet {
+  spec: k8s.V1StatefulSetSpec
+}
+
+interface VolumeSnapshot extends Required<k8s.KubernetesObject> {
+  metadata: {
+    name: string
+    creationTimestamp: Date
+    labels: {
+      statefulset: string
+    }
+  }
+  spec: {
+    volumeSnapshotClassName: string
+    source: {
+      persistentVolumeClaimName: string
+    }
+  }
+}
+
+export interface VolumeReaperArgs {
+  asset: string
+  backupCount: number
+  namespace: string
+  services: string
+}
+
+export class VolumeReaper {
```

Comment:
class reduces the verbose function signatures passing a lot of the same data throughout.

i personally also have a tendency to get lost jumping around too many different files for very associated logic, but that is more personal pref.

i also am not quite as granular with breaking down functions for similar reasons about hopping around too much imo.

just wanted to share so we can get a better understanding and have a little back and  forth on pros/cons, etc.

Reply:
class seems good here. I don't have strong opinions for TS code (...yet), but this looks quite good. Usually I overextract into files and then merge stuff together, easy for me to keep good domain separation in that way. 

This is a good example of that, no need to have these in separate files